### ISSUE TITLE
[Machine Learning] Time Series analysis and forecasting of stock prices using TensorFlow

### ISSUE THREAD

**Issue by jamesomina99** [section-engineering-education/engineering-education#6796]
#### Proposed title of article
[Machine Learning] Time Series analysis and forecasting of stock prices using TensorFlow

#### Proposed article introduction
Stock market prediction is used to determine the future value of company stock or other financial instruments traded on an exchange. The successful prediction of a stock's future price could yield significant profit. Time series analysis comprises methods for analyzing time-series data in order to extract meaningful statistics and other characteristics of the data. 

In investing companies, a time series tracks the movement of the chosen data points, such as a security’s price, over a specified period of time with data points recorded at regular intervals. Time series forecasting fits a model on a historical dataset, that is, the training set, and then using it to predict a future occurrence, which is the test set. 

The performance of the time series model is rated based on its future prediction. We will use TensorFlow to build a Sequential Model with multiple input and hidden layers.  The trained model will be used to predict Apple's future stock prices.

#### Key takeaways 
1. Time series analysis and decomposition of Apple stock data.
2. Initializing the Sequential model.
2. Adding LSTM layers. (multiple hidden layers)
3. Adding Dense Layers.
4. Plotting the actual and prediction prices using Matplotlib

#### Article quality
In this article, we build the time series model using stacked LSTM layers which are more efficient. A Stacked LSTM architecture is an LSTM model that has multiple hidden LSTM layers with each layer containing multiple cells. LSTM models are able to store information over a period of time, this makes them more efficient when training a time series model which is memory intensive.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your topic to the EngEd program. 

After some careful consideration it struck us that this topic may be a bit over saturated throughout other blog sites and official documentations. Please feel free to suggest an alternate topic to explore. 🚀
